### PR TITLE
170 fix issues in vehicle structure

### LINKED ISSUE #148: Origin of vehicle parts
**Is your feature request related to a problem? Please describe.**
Currently we have not defied the origin of the vehicle parts and also the root.

**Describe the solution you'd like**
We previously decided that the main origin for all objects shall be the center of the bounding box in x- and y-direction and the ground in z-direction. So I think this should be the origin of the root.
For the vehicle parts: I think it makes sense to put the origin at the connection point. So the rear part of an articulated bus for example should have its origin in the hinge. Or the cabin of an excavator should have its origin in the attachment (rotation) point to the base.
What I am still uncertain about is the origin of the first vehicle part.

Any ideas @ipg-sig @MatthiasThDs @LudwigFriedmann @MustafaTrian @KimuraDIVP ?

**Describe alternatives you've considered**
As an alternative each vehicle part could also have its origin in its own bounding box center (z on ground). But then we would probably need a separate mechanism to describe the attachment points between different vehicle parts.

### LINKED ISSUE #149: Keywords for vehicle parts
**Is your feature request related to a problem? Please describe.**
Currently the user is completely free to choose any keyword to describe a vehicle part, see [section in documentation](https://asam-ev.github.io/OpenMATERIAL/asamopenmaterial/latest/specification/geometry/object-classes.html#_grp_vehicle_part_part_idx_keyword).

**Describe the solution you'd like**
We should add a list of recommendations for frequently used vehicle parts, e.g. base, cabin, front, rear etc.

Let's collect some ideas here @ipg-sig @MatthiasThDs @LudwigFriedmann @MustafaTrian @KimuraDIVP.

### LINKED ISSUE #170: Fix issues in vehicle structure
**Describe the bug**
See comment in vehicle example PR: https://github.com/asam-ev/OpenMATERIAL/pull/112#issuecomment-2443879623

**Expected behavior**
Fix the issues in the vehicle structure definition

### PR DESCRIPTION
**Pull request by ClemensLinnhoff**
## Describe your changes

- [x] Add mirror joint to side mirrors, blindspot mirror and rearview mirror
- [x] Change all mirrors into a hierarchical structure: mounting -> joint -> view
- [x] Add mirrors to both the exterior_dynamic and the doors on the sides
- [x] Change concept of vehicle parts

## Issue ticket number and link

Fixes #170, fixes #149, fixes #148

## Checklist before requesting a review

- [x] I have performed a self-review of my code/documentation.
- [x] My changes generate no new warnings during the documentation generation.

### CODE REVIEW

**Comment by ClemensLinnhoff** [comment]
@FKlopfer please perform an editorial review and also check, if the node structure is displayed correctly and all links still work, as I changed almost all node names slightly.

**Comment by ClemensLinnhoff** [comment]
@LudwigFriedmann please review, if all changes we discussed in the meeting are contained in this PR.

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [inline content/geometry/object-vehicle/vehicle-structure.adoc:126]:
```suggestion
Typically, this group does not move in the simulation and therefore does not have its own transforms.
```

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [inline content/geometry/object-vehicle/vehicle-structure.adoc:1268]:
```suggestion
The `<side_mirror_view_left_idx>` index numbers each mirror on the left side, counting from right to left in positive y-direction, and from the front to the rear, starting with 0.
```

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [inline content/geometry/object-vehicle/vehicle-structure.adoc:1298]:
```suggestion
The `<side_mirror_view_right_idx>` index numbers each mirror on the right side, counting from right to left in positive y-direction, and from the front to the rear, starting with 0.
```

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [inline content/geometry/object-vehicle/vehicle-structure.adoc:1358]:
```suggestion
Zero rotation and translation around all axles are defined under neutral load conditions.
```

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [inline content/geometry/object-vehicle/vehicle-structure.adoc:1495]:
```suggestion
They may also change their state during the simulation. Examples of dynamic elements are lights, which can be switched on and off.
```

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [inline content/geometry/object-vehicle/vehicle-structure.adoc:1741]:
```suggestion
| Collinear with the vehicle part's longitudinal axis, pointing forwards
```

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [inline content/geometry/object-vehicle/vehicle-structure.adoc:1722]:
```suggestion
A vehicle part is a large component of a vehicle that moves in a slightly different direction than other parts, for example, the vehicle part follows an individual path during turns.
```

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [inline content/geometry/object-vehicle/vehicle-structure.adoc:1088]: I am not sure, whether I understand the sentence correctly. Does the "main moving part" refer to the complete component consisting of the plastic/metal structure that holds the mirror as well as the mirror glass itself? Or are we only talking about the joint, which connects the plastic/metal structure with the vehicle?

My proposals

- A: "This group contains all parts of the movable structure that holds the blindspot mirror. The blindspot mirror automatically adjusts when the blindspot joint is moved."
- B. "This group contains all parts that connect the blindspot mirror mounting to the vehicle. The blindspot mirror automatically adjusts when the angle of the blindspot joint changes."

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [inline content/geometry/object-vehicle/vehicle-structure.adoc:1119]: The modal verb shall is used inconsistently. For example, section Grp_Mirror_Blindspot_Joint_Left uses the following phrase in the same contex:

"It is a child of the corresponding mirror mounting."

For consistency and readability reasons, we should use consistent phrases. We can either remove the shall or, if we we want to emphasize that there are conditions that require different handling, rephrase to:

"If mounted to the vehicle body directly, it is a child node of the group `<Grp_Exterior_Dynamic>`. If mounted to the door it is a child node of `<Grp_Door_Left>` or `<Grp_Door_Right>`."

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [inline content/geometry/object-vehicle/vehicle-structure.adoc:1494]: @ClemensLinnhoff: I rephrased this statement to enhance readability. Please check whether it is correct that lights are treated as dynamic elements because they can change state by switching on/off. Or is the main reason that the driver can adjust the direction of the light emission?

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ClemensLinnhoff) [reply content/geometry/object-vehicle/vehicle-structure.adoc:1088]: Proposal A is correct. It contains all movable parts including the plastic or metal housing and the mirror glass itself (as a child node)

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ClemensLinnhoff) [reply content/geometry/object-vehicle/vehicle-structure.adoc:1089]: Yes that sounds good

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ClemensLinnhoff) [reply content/geometry/object-vehicle/vehicle-structure.adoc:1119]: Yes, let's remove the shall here.

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [inline content/geometry/object-vehicle/vehicle-structure.adoc:196]: In https://github.com/asam-ev/OpenMATERIAL/pull/146/files, we decided that the direction of the z-axis is crucial for the correct direction of the y-axis. This is my proposal for the new z-axis definition:

"Concentric and coaxial to the virtual hinge axis, pointing in the direction that enables the door to move within the xy-plane"

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ClemensLinnhoff) [reply content/geometry/object-vehicle/vehicle-structure.adoc:1494]: Your phrasing is correct, thanks.

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ClemensLinnhoff) [reply content/geometry/object-vehicle/vehicle-structure.adoc:196]: [...] pointing in the direction that enables the door to open with a positive rotation around the z-axis.

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ClemensLinnhoff) [reply content/geometry/object-vehicle/vehicle-structure.adoc:190]: pointing along the closed door sounds good

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [reply content/geometry/object-vehicle/vehicle-structure.adoc:196]: Thanks, I will integrate the feedback today

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [reply content/geometry/object-vehicle/vehicle-structure.adoc]: Integrated feedback

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [reply content/geometry/object-vehicle/vehicle-structure.adoc]: Removed "shall" in this context across file

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [reply content/geometry/object-vehicle/vehicle-structure.adoc]: Updated definition of z-axis across file

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ACMackenthun) [reply content/geometry/object-vehicle/vehicle-structure.adoc]: Updated descriptions across file

**Review by ACMackenthun — APPROVED** [review]

**Comment by FKlopfer** [comment]
@ClemensLinnhoff , as the structure overview in the top has changed, we would have to move the respective chapters to their new places. However, after this it wouldn't be so easy to see in the "files changed"-view what has been changed on this PR regarding the content and what has only been changed regarding the structure. Shall we move the chapters now or wait until the subgroup meeting tomorrow where you can review the changes regarding to the content?

**Comment by ClemensLinnhoff** [comment]
> @ClemensLinnhoff , as the structure overview in the top has changed, we would have to move the respective chapters to their new places. However, after this it wouldn't be so easy to see in the "files changed"-view what has been changed on this PR regarding the content and what has only been changed regarding the structure. Shall we move the chapters now or wait until the subgroup meeting tomorrow where you can review the changes regarding to the content?

@LudwigFriedmann could you review the changes here offline? I think this would be quicker than going over them in the meeting. Then @FKlopfer can make the changes to the structure and we can merge after that.

**Review by LudwigFriedmann — CHANGES_REQUESTED** [review]
I scanned the changes and have identified some minor issues, only.
The only bigger issue I see is the vehicle part conept. Right now, it's ambiguous. I'm not sure if it adds value to the release 1.0 or if it should rather be removed to enable the development of a proper concept in a later release without unnecessary prerequisites.
I'd propose to merge this branch after we've sorted out those topics.
I'll commit a full review on the compiled documentation.

File: content/geometry/object-vehicle/vehicle-general.adoc
Comment (ClemensLinnhoff) [reply content/geometry/object-vehicle/vehicle-general.adoc:11]: Done.

File: content/geometry/object-vehicle/vehicle-general.adoc
Comment (ClemensLinnhoff) [reply content/geometry/object-vehicle/vehicle-general.adoc:14]: Done

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ClemensLinnhoff) [reply content/geometry/object-vehicle/vehicle-structure.adoc:358]: Done

File: content/geometry/object-vehicle/vehicle-structure.adoc
Comment (ClemensLinnhoff) [reply content/geometry/object-vehicle/vehicle-structure.adoc:1804]: I would propose to leave it in, because then people can already gather experience in that direction and bring that into the project for the next OpenMATERIAL version.
But I don't have a strong opinion about it.

Comment (ClemensLinnhoff) [reply content/geometry/object-vehicle/vehicle-structure.adoc:1804]: You need to approve this PR, otherwise I cannot merge, because your change request is still open.

Comment (ClemensLinnhoff) [reply content/geometry/object-vehicle/vehicle-structure.adoc:1804]: Never mind, I could dismiss your change request. Now I can merge.

**Pull request merged**
